vms/platformvm/state/state.go

```diff
@@ -1924,11 +2090,7 @@ func (s *state) GetBlockIDAtHeight(height uint64) (ids.ID, error) {
 	return blkID, nil
 }
 
-func (s *state) writeCurrentStakers(updateValidators bool, height uint64) error {
-	heightBytes := database.PackUInt64(height)
-	rawNestedPublicKeyDiffDB := prefixdb.New(heightBytes, s.nestedValidatorPublicKeyDiffsDB)
-	nestedPKDiffDB := linkeddb.NewDefault(rawNestedPublicKeyDiffDB)
-
+func (s *state) writeCurrentStakers(updateValidators bool) error {
 	for subnetID, validatorDiffs := range s.currentStakers.validatorDiffs {
```

Comment:
1. I feel like now we shouldn't be using `s.currentStakers` and instead be using the returned computed diffs from `processCurrentStakers()`
2. Do we still need this function? Looks like we're applying the weight + key diffs in the other two new functions we're adding